### ISSUE TITLE
Remix Vite cannot run with pnpm on Windows

### ISSUE THREAD

**Issue by hyunrealshadow** [remix-run/remix#8117]
### Reproduction

Clone this [repo](https://github.com/hyunrealshadow/remix-windows-bug), then run `pnpm i`.
Finally, execute `pnpm run build` in the workspace package.

### System Info

```shell
System:
    OS: Windows 11 10.0.22631
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 41.94 GB / 79.93 GB
  Binaries:
    Node: 18.12.1 - ~\scoop\apps\volta\current\appdata\tools\image\node\18.12.1\node.EXE
    Yarn: 1.22.18 - ~\scoop\apps\volta\current\appdata\tools\image\yarn\1.22.18\bin\yarn.CMD
    npm: 9.6.1 - ~\scoop\apps\volta\current\appdata\tools\image\npm\9.6.1\bin\npm.CMD
    pnpm: 8.9.0 - ~\scoop\apps\volta\current\appdata\bin\pnpm.EXE
  Browsers:
    Edge: Chromium (119.0.2151.72)
    Internet Explorer: 11.0.22621.1
```

### Used Package Manager

pnpm

### Expected Behavior

The project can be built successfully.

### Actual Behavior

After executing the command, an error is triggered.
```
[vite]: Rollup failed to resolve import "/@fsE:\project\node_modules\.pnpm\@remix-run+dev@2.3.1_typescript@5.3.2_vite@5.0.2\node_modules\@remix-run\dev\dist\config\defaults\entry.server.node.tsx" from "virtual:server-entry".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
error during build:
Error: [vite]: Rollup failed to resolve import "/@fsE:\project\node_modules\.pnpm\@remix-run+dev@2.3.1_typescript@5.3.2_vite@5.0.2\node_modules\@remix-run\dev\dist\config\defaults\entry.server.node.tsx" from "virtual:server-entry".
This is most likely unintended because it can break your application at runtime.
If you do want to externalize this module explicitly add it to
`build.rollupOptions.external`
```
After debugging, I temporarily modified the code in `node_modules`.
I edited the file https://github.com/remix-run/remix/blob/e5d24b4848d7df41e71a260abb868ee94ca6480b/packages/remix-dev/vite/plugin.ts#L94 and added `/` after `@fs`, which resolved the issue for me.

**Comment by AlemTuzlak** [comment]
@pcattori This is the same error I got that you told me to create an issue on the repo about in discord. You can also reproduce this in Remix Dev Tools:
https://github.com/Code-Forge-Net/Remix-Dev-Tools
- clone the repo
- npm install
- npm run build
- npm run dev:vite

I get the same error

**Comment by hi-ogawa** [comment]
I think this is a duplicate of https://github.com/remix-run/remix/issues/7722#issuecomment-1810263774
This is supposed to be fixed by https://github.com/remix-run/remix/pull/8057 but it seems it was not included in v2.3.1 (even though the bot said it's in v2.3.1-pre.1 https://github.com/remix-run/remix/pull/8057#issuecomment-1823204154).
Could you try nightly?

**Comment by hyunrealshadow** [comment]
> I think this is a duplicate of [#7722 (comment)](https://github.com/remix-run/remix/issues/7722#issuecomment-1810263774) This is supposed to be fixed by #8057 but it seems it was not included in v2.3.1 (even though the bot said it's in v2.3.1-pre.1 [#8057 (comment)](https://github.com/remix-run/remix/pull/8057#issuecomment-1823204154)). Could you try nightly?

I tried the version `@remix-run/dev 0.0.0-nightly-6953c3f-20231123` and it worked for me
Thank You

**Comment by AlemTuzlak** [comment]
@hi-ogawa also can confirm it works now, sweet, this will speed up my remix dev tools vite plugin development by a ton!
